client-akka-http/src/main/scala/bekindrewind/akkahttp/VcrHttp.scala

```diff
@@ -0,0 +1,129 @@
+package bekindrewind.akkahttp
+
+import bekindrewind._
+
+import akka.actor.ClassicActorSystemProvider
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import akka.util.ByteString
+import io.circe.parser._
+import io.circe.syntax._
+
+import java.io.Closeable
+import java.net.URI
+import java.nio.file.Path
+import java.time.OffsetDateTime
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Try
+
+object VcrHttp {
+  def create(
+    sendRequest: HttpRequest => Future[HttpResponse],
+    recordingPath: Path,
+    recordOptions: RecordOptions = RecordOptions.default,
+    matcher: VcrMatcher = VcrMatcher.groupBy(r => (r.method, r.uri))
+  )(implicit executionContext: ExecutionContext, materializer: Materializer): VcrHttp =
+    new VcrHttp(sendRequest, recordingPath, recordOptions, matcher)
+
+  def useClassicActorSystem(
+    recordingPath: Path,
+    recordOptions: RecordOptions = RecordOptions.default,
+    matcher: VcrMatcher = VcrMatcher.groupBy(r => (r.method, r.uri)),
+    executionContext: Option[ExecutionContext] = None
+  )(implicit system: ClassicActorSystemProvider): VcrHttp = {
+    implicit val ec = executionContext.getOrElse(system.classicSystem.dispatcher)
+    create(Http().singleRequest(_), recordingPath, recordOptions, matcher)
+  }
+
+  private[akkahttp] def toVcrRequest(request: HttpRequest): VcrRecordRequest =
+    VcrRecordRequest(
+      method = request.method.value,
+      uri = URI.create(request.uri.toString()),
+      body = request.entity.toString,
+      headers = toVcrHeaders(request.headers)
+    )
+
+  private[akkahttp] def toAkkaResponse(vcrResponse: VcrRecordResponse): HttpResponse = {
+    val store = decode[AkkaHttpResponseStore](vcrResponse.body).toOption.get
+    HttpResponse(
+      status = StatusCode.int2StatusCode(vcrResponse.statusCode),
+      headers = toAkkaHeaders(vcrResponse.headers),
+      entity = HttpEntity.Strict(
+        ContentType.parse(store.contentType).getOrElse(???),
```

Comment:
I think you meant to come back to this unimplemented `???`

Reply:
It is now `throw new IllegalStateException(s"Content-type of a recorded response can not be parsed: ${value}")`.

This could be happen when loading incompatible content-type recorded by other HTTP clients.